### PR TITLE
Update dataset lecture

### PR DESCRIPTION
**Pull request by Enmartz**
This is a new version of the dataset lecture for Colibri repository.

### CODE REVIEW

**Comment by romanjacome99** [comment]
Please create a demo_dataset.py

File: examples/demo_datasets.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+r"""
+Demo Datasets.
+===================================================
+
+In this example we show how to use the custom dataset class to load the predefined datasets in the repository.
+
+"""
+from torch.utils import data
```
Comment (leonsuarez24) [inline examples/demo_datasets.py:8]: Write it below %% so the interactive window works

File: examples/demo_datasets.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# %%
+# Load dataset
+# -----------------------------------------------
+from colibri.data.datasets import CustomDataset
```
Comment (leonsuarez24) [inline examples/demo_datasets.py:13]: add before this line

import os
os.chdir(os.path.dirname(os.getcwd()))

to change the directory of the demo folder to the principal folder

File: examples/demo_datasets.py (hunk 1 above)
Comment (leonsuarez24) [inline examples/demo_datasets.py:8]: add this line after %% so the interactive window can detected when you run the cell
image

File: colibri/data/datasets.py [hunk 3]
```diff
@@ -1,221 +1,89 @@
...
+    def default_transform(self, data):
+        return self.transform_dict['default'](data)
 
-    def load_basic_dataset(self, name):
-        """
-        Load basic data with labels or classes.
 
-        Args:
-            name (str): Name of the dataset (mnist, fashion_mnist, cifar10, cifar100).
+class CustomDataset(Dataset):
```
Comment (leonsuarez24) [inline colibri/data/datasets.py:36]: add in this documentation all the available datasets ( 'mnist',  'fashion_mnist', 'cifar10', 'cifar100') to avoid users looking in other files what are the available datasets 

also add that there is available  readers for cave and arad datasets

**Comment by leonsuarez24** [comment]
update datasets.rst and mention there the available datasets

File: examples/demo_colibri.py [hunk 4]
```diff
@@ -46,28 +47,33 @@
...
-
 # %%
 # Load dataset
 # -----------------------------------------------
-from colibri.data.datasets import Dataset
+from colibri.data.datasets import CustomDataset
 
-dataset_path = 'cifar10'
-keys = ''
+name = 'cifar10'
```
Comment (leonsuarez24) [inline examples/demo_colibri.py:55]: add a comment with the available datasets similar to the one you mention belowe to the adquisicon systems (adquistion_name = 'c_cassi'  # ['spc', 'cassi'])

**Comment by leonsuarez24** [comment]
maybe in the demo you could also show how the data readers work for not built in datasets

File: colibri/data/datasets.py (hunk 3 above)
Comment (Enmartz) [reply colibri/data/datasets.py:67]: solved

File: examples/demo_colibri.py (hunk 4 above)
Comment (Enmartz) [reply examples/demo_colibri.py:55]: solved

File: examples/demo_datasets.py (hunk 1 above)
Comment (Enmartz) [reply examples/demo_datasets.py:13]: solved

File: examples/demo_datasets.py (hunk 2 above)
Comment (Enmartz) [reply examples/demo_datasets.py:21]: solved

**Comment by Enmartz** [comment]
> maybe in the demo you could also show how the data readers work for not built in datasets

The cave dataset was added to the demo example

File: examples/demo_datasets.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+    plt.subplot(3, 3, (3 * i) + 1)
+    plt.imshow(rgb_image[i].permute(1, 2, 0).cpu().numpy())
+    plt.title('rgb')
+    plt.axis('off')
+
+    plt.subplot(3, 3, (3 * i) + 2)
+    plt.imshow(spec_image[i, [18, 12, 8]].permute(1, 2, 0).cpu().numpy())
+    plt.scatter(coord1[1], coord1[0], s=120, edgecolors='black')
+    plt.scatter(coord2[1], coord2[0], s=120, edgecolors='black')
```
Comment (leonsuarez24) [inline examples/demo_datasets.py:119]: maybe you could add axis titles to the signature plot

**Comment by leonsuarez24** [comment]
couldn't find in the docs 
image

**Comment by leonsuarez24** [comment]
i think it would be great if you could include in the docs the documentation of the class CustomDataset

**Comment by Enmartz** [comment]
> couldn't find BASIC_DATASETS in the docs image

Solved!

**Review by bemc22 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ You can go to ``examples`` folder and run cells of the notebook ``demo_colibri.i
 
 ```python
 import matplotlib.pyplot as plt
-from colibri.data.datasets import Dataset
+from colibri.data.datasets_ import Dataset
```
[the remark this answers is not in the document]
Comment (Enmartz) [reply README.md:50]: solved

File: README.md [hunk 7]
```diff
@@ -57,7 +57,7 @@ keys = ''
 batch_size = 128
 
 dataset = Dataset(dataset_path, keys, batch_size)
-adquistion_name = 'cassi' #  ['spc', 'cassi']
+adquistion_name = 'cassi'  # ['spc', 'cassi']
```
[the remark this answers is not in the document]
Comment (Enmartz) [reply README.md:60]: solved

File: colibri/data/datasets.py [hunk 8]
```diff
@@ -1,221 +1,117 @@
...
-            with h5py.File(str(filename), 'r') as mat:
-                spectral_image = np.float32(np.array(mat[self.keys['spec']]))
-                spectral_image = np.transpose(spectral_image, (2, 1, 0))
-
-            filename = filename.replace('_spectral', '_RGB')
-            image = load_image(filename.replace('.mat', '.jpg'))
-
-            spectral_image = torch.from_numpy(spectral_image.copy()).permute(2, 0, 1)
-            image = torch.from_numpy(image.copy()).permute(2, 0, 1)
+DATASET_READER = {
```
[the remark this answers is not in the document]
Comment (Enmartz) [reply colibri/data/datasets.py:11]: `D` are the functions that come from `data/utils.py`. You can find it at the beginning of this script:

image

**Review by bemc22 — APPROVED** [review]
